### ISSUE TITLE
Add Scope to a Custom dimension

### ISSUE THREAD

**Issue by jwiel86** [tombenner/reports_kit#14]
Awesome project. 

I need to group my x-axis by month which is proving difficult.  Is there a way to do a custom dimension grouped by a scope?

Here's what I have:

.yml file:

    dimensions:
    - aggregate_by_month

model file:

    scope :by_month, -> { select('st_event_logs.*, st_event_logs.session_date_time as session_date_time').group_by{ |x| x.session_date_time.beginning_of_month } }

      include ReportsKit::Model
      reports_kit do 
        dimension :aggregate_by_month, group: "by_month"
      end
    end

**Comment by jwiel86** [comment]
So I noticed here:  https://www.reportskit.co/subcategories/first_day_of_the_week

that there's a way to change how you group by week, is there a way to group by month?

**Comment by tombenner** [comment]
tl;dr: If you use the master branch of ReportsKit, the following should work:

```yaml
dimensions:
- key: session_date_time
  granularity: month
```

Support for a month granularity in a dimension was just added in 31826ad0080e6d8492d6064e8a8a0797c690acfc. There are examples of using either a `month` granularity or a `day` granularity in [these tests](https://github.com/tombenner/reports_kit/blob/master/spec/reports_kit/reports/data/generate_spec.rb#L247-L306), but I haven't added documentation for it yet. I'll try to do that soon, but if anyone else wants to add it, it would be documented in [tombenner/reports_kit_docs](https://github.com/tombenner/reports_kit_docs#contributing).

This support will be available in the next release of ReportsKit, but until then, you'll need to use the master branch of this repo.

**Comment by jwiel86** [comment]
This worked beautifully!  Thank you so much!
